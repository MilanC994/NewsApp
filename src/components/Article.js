import React from 'react'
import { Container, Image, Card, Figure, ListGroup, Row, Col, Badge } from 'react-bootstrap'
import { Redirect, Link } from 'react-router-dom'
import { dummyText } from '../utils/constants'
import { authorSvg, dateSvg, sourceSvg } from '../utils/svgIcons'
import moment from 'moment'

const Article = (props) => {
    if(!props.location.state)
        return <Redirect to="/"/>
    const { title, source, author, description, urlToImage, publishedAt, content, url } = props.location.state

    const datePublishedAt = publishedAt ? moment(publishedAt).format('Do MMMM YYYY HH:SS') : 'Date not provided'
    const sourceName = source && source.name ? source.name : 'Source Not Provided'
    return (
    <Container>
        <Card className="article-card">
            <h1 className="display-5">{title}</h1>
            <Card.Header><Card.Subtitle className="display-5">{description}</Card.Subtitle></Card.Header>
            <Card.Body>
                
                <Row style={{marginTop:"10px"}}>
                    <Col>
                        <ListGroup  horizontal>
                            <ListGroup.Item variant="light" >{authorSvg()} Author:{author ? author:'Not Listed'}</ListGroup.Item>
                        </ListGroup> 
                    </Col>
                    <Col bsPrefix="justify">
                        <ListGroup  horizontal >
                            <ListGroup.Item variant="light" >{dateSvg()} Published At:{datePublishedAt}</ListGroup.Item>
                        </ListGroup>
                    </Col>
                </Row>
                <hr/>
                <Container>
                    <Image alt="article-image" style={{ float:"left", margin:"10px", maxHeight:"800px"}} src={urlToImage} width="100%"/>
                    {content ? <Card.Text>{content}{dummyText}</Card.Text> : <Card.Text>Content Not Provided. Click <Link to="/">here</Link> to go to front page</Card.Text>}
                </Container>
                <hr/>
                <Row style={{marginTop:"10px"}}>
                    <Col>
                        <ListGroup  horizontal>
                            <ListGroup.Item variant="light" >{sourceSvg()} Source:{sourceName}</ListGroup.Item>
                        </ListGroup> 
                    </Col>
                </Row>
            </Card.Body>
            <Card.Footer>
            <Row style={{marginTop:"10px"}}>
                    <Col>
                        <ListGroup  horizontal>
                            <ListGroup.Item variant="light" ><Link to="/">Home Page</Link></ListGroup.Item>
                        </ListGroup> 
                    </Col>
                    <Col  bsPrefix="justify"> 
                        <ListGroup  horizontal style={{width:"fit-content"}}>
                            <ListGroup.Item variant="light"><a href={url} target="_blank" rel="noopener noreferrer" >Visit Source Website</a></ListGroup.Item>
                        </ListGroup>
                    </Col>
                </Row>
            </Card.Footer>
        </Card> 
     </Container>
    )
}

export default Article
