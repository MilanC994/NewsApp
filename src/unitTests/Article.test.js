import React from 'react'
import { mount } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import { mountToJson } from 'enzyme-to-json'
import Article from '../components/Article'

describe('Article   test', () => {
  test('Renders correctly', () => {
    const componentProps = {
      location: {
        state: {
          title: 'test Title',
          source: 'testSource',
          author: 'testAuthor',
          description: 'testDescription',
          urlToImage: 'testUrlToImage',
          publishedAt: '2021-01-02T13:25:54Z',
          content: 'test article content',
          url: 'testurl',
        },
      },
    }
    const wrapper = mount(
      <Router>
        <Article {...componentProps} />
      </Router>,
    ).childAt(0)
    expect(mountToJson(wrapper)).toMatchSnapshot()
  })
  test('Does not render component when location props is null', () => {
    const wrapper = mount(
      <Router>
        <Article />
      </Router>,
    ).childAt(0)
    const element = wrapper.find('.article-card')
    expect(element.exists()).toBeFalsy()
  })
})
