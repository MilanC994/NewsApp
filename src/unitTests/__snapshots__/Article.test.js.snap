// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article   test Renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Article
    location={
      Object {
        "state": Object {
          "author": "testAuthor",
          "content": "test article content",
          "description": "testDescription",
          "publishedAt": "2021-01-02T13:25:54Z",
          "source": "testSource",
          "title": "test Title",
          "url": "testurl",
          "urlToImage": "testUrlToImage",
        },
      }
    }
  >
    <Container
      fluid={false}
    >
      <div
        className="container"
      >
        <Card
          body={false}
          className="article-card"
        >
          <div
            className="article-card card"
          >
            <h1
              className="display-5"
            >
              test Title
            </h1>
            <CardBody>
              <div
                className="card-body"
              >
                <ListGroup
                  horizontal="sm"
                >
                  <ForwardRef
                    as="div"
                    className="list-group list-group-horizontal-sm"
                    onSelect={[Function]}
                  >
                    <div
                      className="list-group list-group-horizontal-sm"
                      onKeyDown={[Function]}
                    >
                      <ListGroupItem
                        active={false}
                        disabled={false}
                        variant="dark"
                      >
                        <ForwardRef
                          as="div"
                          className="list-group-item list-group-item-dark"
                          disabled={false}
                          eventKey={null}
                          onClick={[Function]}
                        >
                          <div
                            className="list-group-item list-group-item-dark"
                            data-rb-event-key={null}
                            disabled={false}
                            onClick={[Function]}
                          >
                            <svg
                              className="bi bi-people"
                              fill="currentColor"
                              height="16"
                              viewBox="0 0 16 16"
                              width="16"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M15 14s1 0 1-1-1-4-5-4-5 3-5 4 1 1 1 1h8zm-7.978-1A.261.261 0 0 1 7 12.996c.001-.264.167-1.03.76-1.72C8.312 10.629 9.282 10 11 10c1.717 0 2.687.63 3.24 1.276.593.69.758 1.457.76 1.72l-.008.002a.274.274 0 0 1-.014.002H7.022zM11 7a2 2 0 1 0 0-4 2 2 0 0 0 0 4zm3-2a3 3 0 1 1-6 0 3 3 0 0 1 6 0zM6.936 9.28a5.88 5.88 0 0 0-1.23-.247A7.35 7.35 0 0 0 5 9c-4 0-5 3-5 4 0 .667.333 1 1 1h4.216A2.238 2.238 0 0 1 5 13c0-1.01.377-2.042 1.09-2.904.243-.294.526-.569.846-.816zM4.92 10A5.493 5.493 0 0 0 4 13H1c0-.26.164-1.03.76-1.724.545-.636 1.492-1.256 3.16-1.275zM1.5 5.5a3 3 0 1 1 6 0 3 3 0 0 1-6 0zm3-2a2 2 0 1 0 0 4 2 2 0 0 0 0-4z"
                              />
                            </svg>
                             Author: 
                            testAuthor
                          </div>
                        </ForwardRef>
                      </ListGroupItem>
                      <ListGroupItem
                        active={false}
                        disabled={false}
                        variant="dark"
                      >
                        <ForwardRef
                          as="div"
                          className="list-group-item list-group-item-dark"
                          disabled={false}
                          eventKey={null}
                          onClick={[Function]}
                        >
                          <div
                            className="list-group-item list-group-item-dark"
                            data-rb-event-key={null}
                            disabled={false}
                            onClick={[Function]}
                          >
                            <svg
                              className="bi bi-pencil-fill"
                              fill="currentColor"
                              height="16"
                              viewBox="0 0 16 16"
                              width="16"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"
                              />
                            </svg>
                             Source: 
                            Source Not Provided
                          </div>
                        </ForwardRef>
                      </ListGroupItem>
                      <ListGroupItem
                        active={false}
                        disabled={false}
                        variant="dark"
                      >
                        <ForwardRef
                          as="div"
                          className="list-group-item list-group-item-dark"
                          disabled={false}
                          eventKey={null}
                          onClick={[Function]}
                        >
                          <div
                            className="list-group-item list-group-item-dark"
                            data-rb-event-key={null}
                            disabled={false}
                            onClick={[Function]}
                          >
                            <svg
                              className="bi bi-calendar-date-fill"
                              fill="currentColor"
                              height="16"
                              viewBox="0 0 16 16"
                              width="16"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M4 .5a.5.5 0 0 0-1 0V1H2a2 2 0 0 0-2 2v1h16V3a2 2 0 0 0-2-2h-1V.5a.5.5 0 0 0-1 0V1H4V.5zm5.402 9.746c.625 0 1.184-.484 1.184-1.18 0-.832-.527-1.23-1.16-1.23-.586 0-1.168.387-1.168 1.21 0 .817.543 1.2 1.144 1.2z"
                              />
                              <path
                                d="M16 14V5H0v9a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2zm-6.664-1.21c-1.11 0-1.656-.767-1.703-1.407h.683c.043.37.387.82 1.051.82.844 0 1.301-.848 1.305-2.164h-.027c-.153.414-.637.79-1.383.79-.852 0-1.676-.61-1.676-1.77 0-1.137.871-1.809 1.797-1.809 1.172 0 1.953.734 1.953 2.668 0 1.805-.742 2.871-2 2.871zm-2.89-5.435v5.332H5.77V8.079h-.012c-.29.156-.883.52-1.258.777V8.16a12.6 12.6 0 0 1 1.313-.805h.632z"
                              />
                            </svg>
                             Published At: 
                            2nd January 2021 14:00
                          </div>
                        </ForwardRef>
                      </ListGroupItem>
                    </div>
                  </ForwardRef>
                </ListGroup>
                <hr />
                <CardSubtitle
                  className="display-5"
                >
                  <ForwardRef
                    className="display-5 card-subtitle"
                  >
                    <div
                      className="display-5 card-subtitle h6"
                    >
                      testDescription
                    </div>
                  </ForwardRef>
                </CardSubtitle>
                <hr />
                <Container
                  fluid={false}
                >
                  <div
                    className="container"
                  >
                    <Figure>
                      <figure
                        className="figure"
                      >
                        <FigureImage
                          alt="article-image"
                          fluid={true}
                          src="testUrlToImage"
                          style={
                            Object {
                              "float": "left",
                              "margin": "10px",
                              "maxHeight": "800px",
                            }
                          }
                          width="100%"
                        >
                          <Image
                            alt="article-image"
                            className="figure-img"
                            fluid={true}
                            rounded={false}
                            roundedCircle={false}
                            src="testUrlToImage"
                            style={
                              Object {
                                "float": "left",
                                "margin": "10px",
                                "maxHeight": "800px",
                              }
                            }
                            thumbnail={false}
                            width="100%"
                          >
                            <img
                              alt="article-image"
                              className="figure-img img-fluid"
                              src="testUrlToImage"
                              style={
                                Object {
                                  "float": "left",
                                  "margin": "10px",
                                  "maxHeight": "800px",
                                }
                              }
                              width="100%"
                            />
                          </Image>
                        </FigureImage>
                        <FigureCaption>
                          <figcaption
                            className="figure-caption"
                          >
                            Nulla vitae elit libero, a pharetra augue mollis interdum.
                          </figcaption>
                        </FigureCaption>
                      </figure>
                    </Figure>
                    <CardText
                      align="justify"
                      className="content-paragraph"
                    >
                      <p
                        align="justify"
                        className="content-paragraph card-text"
                      >
                        test article content
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut laboreet dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecatcupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut laboreet dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecatcupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut laboreet dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecatcupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                      </p>
                    </CardText>
                  </div>
                </Container>
              </div>
            </CardBody>
            <CardFooter>
              <div
                className="card-footer"
              >
                <Row
                  noGutters={false}
                  style={
                    Object {
                      "marginTop": "10px",
                    }
                  }
                >
                  <div
                    className="row"
                    style={
                      Object {
                        "marginTop": "10px",
                      }
                    }
                  >
                    <Col>
                      <div
                        className="col"
                      >
                        <ListGroup
                          horizontal={true}
                        >
                          <ForwardRef
                            as="div"
                            className="list-group list-group-horizontal"
                            onSelect={[Function]}
                          >
                            <div
                              className="list-group list-group-horizontal"
                              onKeyDown={[Function]}
                            >
                              <ListGroupItem
                                active={false}
                                disabled={false}
                                variant="light"
                              >
                                <ForwardRef
                                  as="div"
                                  className="list-group-item list-group-item-light"
                                  disabled={false}
                                  eventKey={null}
                                  onClick={[Function]}
                                >
                                  <div
                                    className="list-group-item list-group-item-light"
                                    data-rb-event-key={null}
                                    disabled={false}
                                    onClick={[Function]}
                                  >
                                    <Link
                                      to="/"
                                    >
                                      <LinkAnchor
                                        href="/"
                                        navigate={[Function]}
                                      >
                                        <a
                                          href="/"
                                          onClick={[Function]}
                                        >
                                          Home Page
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </div>
                                </ForwardRef>
                              </ListGroupItem>
                            </div>
                          </ForwardRef>
                        </ListGroup>
                      </div>
                    </Col>
                    <Col
                      bsPrefix="justify"
                    >
                      <div
                        className="justify"
                      >
                        <ListGroup
                          horizontal={true}
                          style={
                            Object {
                              "width": "fit-content",
                            }
                          }
                        >
                          <ForwardRef
                            as="div"
                            className="list-group list-group-horizontal"
                            onSelect={[Function]}
                            style={
                              Object {
                                "width": "fit-content",
                              }
                            }
                          >
                            <div
                              className="list-group list-group-horizontal"
                              onKeyDown={[Function]}
                              style={
                                Object {
                                  "width": "fit-content",
                                }
                              }
                            >
                              <ListGroupItem
                                active={false}
                                disabled={false}
                                variant="light"
                              >
                                <ForwardRef
                                  as="div"
                                  className="list-group-item list-group-item-light"
                                  disabled={false}
                                  eventKey={null}
                                  onClick={[Function]}
                                >
                                  <div
                                    className="list-group-item list-group-item-light"
                                    data-rb-event-key={null}
                                    disabled={false}
                                    onClick={[Function]}
                                  >
                                    <a
                                      href="testurl"
                                      rel="noopener noreferrer"
                                      target="_blank"
                                    >
                                      Visit Source Website
                                    </a>
                                  </div>
                                </ForwardRef>
                              </ListGroupItem>
                            </div>
                          </ForwardRef>
                        </ListGroup>
                      </div>
                    </Col>
                  </div>
                </Row>
              </div>
            </CardFooter>
          </div>
        </Card>
      </div>
    </Container>
  </Article>
</Router>
`;
